oeps/oep-0042-arch-authentication.rst

```diff
@@ -0,0 +1,254 @@
+
+.. _ADR on Django OAuth Toolkit (DOT): https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/docs/decisions/0002-migrate-to-dot.rst
+.. _ADR to use JWTs as OAuth Tokens: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/docs/decisions/0003-use-jwt-as-oauth-tokens-remove-openid-connect.rst
+.. _ADR on Asymmetric JWTs: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/docs/decisions/0008-use-asymmetric-jwts.rst
+.. _JWT Cookies ADR: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/docs/decisions/0009-jwt-in-session-cookie.rst
+.. _oauth_dispatch (edx-platform): https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/docs/README.rst
+.. _oauth_dispatch as router ADR: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/docs/decisions/0004-oauth-dispatch-as-router.rst#L33
+
+
+OAuth2 and Bearer Tokens
+------------------------
+
+This section refers to Bearer Tokens as documented in `OAuth 2.0 RFC: Bearer Token Usage`_.
+
+Currently, only the mobile applications (iOS and Android) should continue to use Bearer Tokens. In this method, the LMS issues a token to the mobile apps and the mobile apps include this token in their header when hitting IDA endpoints. The IDAs verify the token and either service or reject the request, based on the token validity.
+
+The communications between the browser, LMS, and IDA must all use TLS in order to keep the token secure, as anyone with the token can make a restricted IDA request.
+
+All other usage of Bearer Tokens in Open edX has been deprecated. Mobile applications may one day move to JWTs as well, but that is a larger effort.
+
+.. _`OAuth 2.0 RFC: Bearer Token Usage`: https://tools.ietf.org/html/rfc6750
+
+Social (and Other) Authentication
+---------------------------------
+
+Open edX platform also supports several social authentication methods, such as Google, Facebook, and LinkedIn, along with other campus/business-specific authentication methods, including LTI and SAML. These external authentication methods are used to integrate or link your edX identity to another network identity. However, once the identity link is established and an Open edX account is created, the LMS still functions as usual as the idP for all satellite IDAs, and the authentication method remains the same.
+
+Code handling for LTI/SAML authentication, as well as python-social-auth login, such as Google, Facebook, etc., can be found in `third_party_auth (edx-platform)`. This implementation is supported by the `python-social-auth library`_.
+
+.. _third_party_auth (edx-platform): https://github.com/edx/edx-platform/tree/master/common/djangoapps/third_party_auth
+.. _python-social-auth library: https://github.com/omab/python-social-auth
+
+OAuth2 and Authorization
```

Comment:
Only `OEP-4` is related to `OAuth2`.  None of the other OEPs listed in this section are related to OAuth2. I find it confusing to list them in an OEP specific for Authentication.  From SRP perspective,  it would be good to limit this OEP to its single axes of change - whenever authentication changes (and not need to update this OEP when authorization changes).

Given that, can we just cross-reference to OEP-4 or leave this section out altogether?

Reply:
For now, I moved this list of Authorization decisions as notes in [ARCHBOM-1216](https://openedx.atlassian.net/browse/ARCHBOM-1216) for an Authorization OEP.